resharper/resharper-unity/src/Rider/UnitTesting/RunViaUnityEditorStrategy.cs

```diff
@@ -481,43 +475,49 @@ private Task WaitForUnityEditorConnectedAndIdle(Lifetime lifetime)
             return JetTaskEx.While(() => waitingLifetime.IsAlive);
         }
 
-        private IEnumerable<IUnitTestElement> CollectElementsToRunInUnityEditor(IUnitTestRun firstRun)
+        private List<TestFilter> GetFilters(IUnitTestRun firstRun)
         {
-            var result = new JetHashSet<IUnitTestElement>();
-            foreach (var unitTestRun in firstRun.Launch.Runs)
+            var filters = new List<TestFilter>();
+
+            var runs = firstRun.Launch.Runs;
+            foreach (var run in runs)
             {
-                if (unitTestRun.RunStrategy.Equals(this))
+                if (!run.RunStrategy.Equals(this))
+                    continue;
+                
+                var unitTestElements = new JetHashSet<IUnitTestElement>();
+                unitTestElements.AddRange(run.Elements);
+                var elements = unitTestElements
+                    .Where(unitTestElement => unitTestElement is NUnitTestElement ||
+                                              unitTestElement is NUnitRowTestElement).ToArray();
+
+                var testNames = elements.Select(p => p.Id.Id).ToList();
+                
+                var groups = new List<string>();
+                var categories = new List<string>();
+                var criterion = run.Launch.Criterion.Criterion;
+                if (criterion is ConjunctiveCriterion conjunctiveCriterion)
                 {
-                    result.AddRange(unitTestRun.Elements);
+                    groups.AddRange( conjunctiveCriterion.Criteria.Where(a => a is TestAncestorCriterion).SelectMany(b =>
```

Comment:
I still recommend you not to inspect criterion yourself. Instead, use `run.Elements` properties. This property returns the list of all elements that are to be executed by this run. If you want to exclude explicit elements, you can do it based on the value of `NUnitElementBase.Explicit` property. The resulting filtered list you only need to convert it into a group filter, and that's it.
The way it is written now, you are at the very least not taking into account different rules that might be inplace. For example, `RemoveElementsWithIgnoredCategoriesRule`.